docs/guides/circuits.rst

```diff
@@ -0,0 +1,439 @@
+  ic| np.allclose(inverse_unitary, np.eye(8)): True
+
+
+.. _appending_circuits:
+
+Appending operations to a circuit
+---------------------------------
+
+You don't have to build a whole circuit all at once! Let's see how to append operations to an existing circuit by constructing the next part of our QAOA circuit, the problem hamiltonian circuit. This problem hamiltonian circuit gives a quantum description of the problem and shows how the nodes are connected. In this case, the 3 nodes are all-to-all connected, so our circuit should look like this:
+
+.. code-block:: text                                                                     
+                                                                                
+  q_0: ──■───────────────────■────■───────────────────■───────────────────────────
+       ┌─┴─┐┌─────────────┐┌─┴─┐  │                   │                           
+  q_1: ┤ X ├┤ Rz($\beta$) ├┤ X ├──┼───────────────────┼────■───────────────────■──
+       └───┘└─────────────┘└───┘┌─┴─┐┌─────────────┐┌─┴─┐┌─┴─┐┌─────────────┐┌─┴─┐
+  q_2: ─────────────────────────┤ X ├┤ Rz($\beta$) ├┤ X ├┤ X ├┤ Rz($\beta$) ├┤ X ├
+                                └───┘└─────────────┘└───┘└───┘└─────────────┘└───┘
+
+We'll build up this circuit one connection at a time, using ``+=`` to append ``Circuit`` objects to the end of our existing ``problem_hamiltonian_circ``
+
+.. literalinclude:: /examples/circuits_guide.py
+  :language: python
+  :start-at: problem_hamiltonian_circ = Circuit(
+  :end-before: unitary = problem_hamiltonian_circ.to_unitary()
+
+For now, don't worry about what the ``beta`` variables mean in these circuits. We'll cover that later when we talk about :ref:`symbolic gates <symbolic_gates>`. For now, just notice that we started with an initial circuit and we were able to append other circuits to it using ``+=``
+
+Let's get the :ref:`unitary matrix <getting_unitary>` again and see if it makes sense for this circuit
+
+.. literalinclude:: /examples/circuits_guide.py
+  :language: python
+  :start-at: unitary = problem_hamiltonian_circ.to_unitary()
+  :end-at: ic(type(unitary))
+
+The output should look like this:
+
+.. code-block:: text
+
+  ic| unitary.shape: (8, 8)
+  ic| type(unitary): <class 'sympy.matrices.immutable.ImmutableDenseMatrix'>
+
+For more info on why this is a sympy matrix as opposed to a numpy ndarray, please see the section on :ref:`symbolic gates <symbolic_gates>`.
+
+Appending individual gates to circuits is also possible, as we'll see when we build up the mixing circuit for this QAOA problem. The mixing circuit looks like this:
+
+.. code-block:: text
+
+       ┌──────────────┐
+  q_0: ┤ Rx($\gamma$) ├
+       ├──────────────┤
+  q_1: ┤ Rx($\gamma$) ├
+       ├──────────────┤
+  q_2: ┤ Rx($\gamma$) ├
+       └──────────────┘
+
+To build up the mixing circuit, we ust need to put parametrized RX gate on each of the qubits. We can do that with a for loop:
```

Comment:
Might be worth mentioning that we have a utility tool that does exactly that as well :) 
https://github.com/zapatacomputing/orquestra-quantum/blob/f58c2ffdba16cef4f4894dea8521c00de44a9797/src/orquestra/quantum/circuits/_generators.py#L14

Reply:
fixed in af446a1749110a289630084291990551053306a0